crates/clients/matchmaker/src/client.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::Arc;
+
+use ethers::{signers::Signer, types::Chain};
+
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::core::Error as RpcError;
+use jsonrpsee::http_client::{transport::HttpBackend, HttpClient, HttpClientBuilder};
+
+use tower::ServiceBuilder;
+
+use crate::{
+    flashbots_signer::{FlashbotsSigner, FlashbotsSignerLayer},
+    types::{BundleRequest, SendBundleResponse},
+};
+
+/// Matchmaker client to interact with MEV-share
+pub struct Client<S> {
+    /// Underlying HTTP client
+    pub http_client: HttpClient<FlashbotsSigner<S, HttpBackend>>,
+}
```

Comment:
This is :+1: given you use jsonrpsee, you could also have done it this way: https://github.com/onbjerg/ethers-flashbots/blob/master/src/relay.rs#L83-L101, just using `reqwest`, but maybe you had other reasons not to?

Reply:
yep! Writing the tower middleware felt a bit cleaner since it allowed us to take advantage of jsonrpsee instead of copying in snippets of json-rpc code, but both work